zcash_client_sqlite/src/lib.rs

```diff
@@ -1269,6 +1292,11 @@ impl<P: consensus::Parameters> WalletWrite for WalletDb<rusqlite::Connection, P>
                         .enumerate()
                     {
                         if let Some(address) = txout.recipient_address() {
+                            // TODO: we really want to only mark outputs when a transaction has been
+                            // *reliably* mined.
```

Comment:
Per my previous comment, I think I disagree? Or rather, if we're a decent way away from the gap limit then I think we want to detect any usage of the ephemeral addresses, even if it doesn't end up getting mined. If we're getting close to the gap limit OTOH then we probably need to only mark reliably-mined transactions (or unmark on reorgs), to avoid going over the gap limit and breaking TEX address sending entirely. Hmm...

Reply:
It is more conservative to only mark outputs when a transaction has been reliably mined. Remember that we will only detect use of ephemeral addresses that are in the database, and we never reuse those in a new transaction anyway. We use the fact that an addresses has been observed in a mined transaction only to extend the gap. If we refrain from doing so until it has been reliably mined, then we will get to the end of the gap sooner.